src/test/groovy/com/palantir/gradle/versions/ConfigurationOnDemandSpec.groovy

```diff
@@ -0,0 +1,423 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.Unroll
+
+import static com.palantir.gradle.versions.GradleTestVersions.GRADLE_VERSIONS
+import static com.palantir.gradle.versions.PomUtils.makePlatformPom
+
+/**
+ * This tests the interaction of this plugin with Gradle's configuration-on-demand feature:
+ * https://docs.gradle.org/current/userguide/multi_project_configuration_and_execution.html#sec:configuration_on_demand
+ */
+@Unroll
+class ConfigurationOnDemandSpec extends IntegrationSpec {
```

Comment:
These look like great tests - very readable - thanks for writing them!

Strangely I can't run them without getting `java.lang.ClassNotFoundException: com.google.common.collect.Multimap`, but that's also true on develop with other tests, so probably a me problem.

Reply:
I personally ran into some problems with "Could not load compiled classes for build file" when running tests locally, but switching the Gradle wrapper to use a non-snapshot version fixed that. (Could be a security software interaction, per [this discussion](https://discuss.gradle.org/t/gradleexception-could-not-load-compiled-classes-for-build-file-fails-on-all-scripts-even-after-removing-cache/2178/5).)